source/game.game.bmx

```diff
@@ -211,7 +211,7 @@ Type TGame Extends TGameBase {_exposeToLua="selected"}
 		EndIf
 
 
-		Local timeData:TData = New TData.AddInt("minute", GetWorldTime().GetDayMinute()).AddInt("hour", GetWorldTime().GetDayHour()).AddInt("day", GetWorldTime().GetDay())
+		Local timeData:TData = New TData.AddInt("minute", GetWorldTime().GetDayMinute()).AddInt("hour", GetWorldTime().GetDayHour()).AddInt("day", GetWorldTime().GetDay()).AddLong("time", GetWorldTime().GetTimeGone())
```

Comment:
Bei den häufig aufgerufenen Zeitumrechnungsmethoden frage ich mich manchmal, ob das einen negativen Einfluss hat...
Schließlich läuft die Zeit immer nur vorwärts.

Reply:
Die Getter sollte der Compiler eh "inlinen" ... du kannst natuerlich fuer die ganzen spezifischen Anfragen auch "field" anlegen und diese beim aktualisieren mit neuen Werten fuellen - und dann direkt abrufen.

Reply:
Zumindest in der KI frage ich mich, ob das für Tag, Stunde, Minute etc. gemacht werden sollte, damit die KI nicht immer aus Lua raus aufrufen muss.

Reply:
wenn die Events asynchron (eigene threads) bzw parallel zum Spiel abgearbeitet werden, muss aber die Zeitinformation vo n "damals" mit uebergeben werden. Vielleicht nicht unbedingt im "minute/hour/..."-Detailverliebtheitsgrad aber irgendwo schon (notfalls als "time:long").

Reply:
Wenn man zum Aufrufzeitpunkt TVT.getDayMinute() aufruft, um eine Abfrage zu machen, ist das ja auch nicht die Zeit von damals - da kann man also genausogut die "gespeicherte" Zeit nehmen.

Reply:
Die Optimierung von den "worldtime"-Abfragen koenntest Du gern in einem neuen Issue "aufheben" ... dann gehen wir das bei Gelegenheit mit an.